source/s3_meta_request.c

```diff
@@ -1145,6 +1145,18 @@ void aws_s3_meta_request_finish_default(struct aws_s3_meta_request *meta_request
         aws_s3_request_release(release_request);
     }
 
+    if (meta_request->headers_callback && finish_result.error_response_headers) {
+        if (meta_request->headers_callback(
+                meta_request,
+                finish_result.error_response_headers,
+                finish_result.response_status,
+                meta_request->user_data)) {
+            /* TODO: decide whether to override the existing error code */
```

Comment:
oops, I put the todo just as I don't know which is better. This code is override the error code using the one from callback, I feel like it's better when I wrote it. If you agree, we can just remove this todo.

Reply:
Removed TODO, thanks!